src/value/mod.rs

```diff
@@ -167,6 +180,43 @@ pub(crate) fn mg_value_string(mg_value: *const bindings::mg_value) -> String {
     mg_string_to_string(c_str)
 }
 
+pub(crate) fn mg_value_naive_date(mg_value: *const bindings::mg_value) -> Option<NaiveDate> {
+    let c_date = unsafe { bindings::mg_value_date(mg_value) };
+    let c_delta_days = unsafe { bindings::mg_date_days(c_date) };
+    let epoch_date = NaiveDate::from_ymd(1970, 1, 1);
+    let delta_days = Duration::days(c_delta_days);
+    epoch_date.checked_add_signed(delta_days)
+}
+
+pub(crate) fn mg_value_naive_local_time(mg_value: *const bindings::mg_value) -> Option<NaiveTime> {
+    let nsec_in_sec = 1_000_000_000;
+    let c_local_time = unsafe { bindings::mg_value_local_time(mg_value) };
+    let c_nanoseconds = unsafe { bindings::mg_local_time_nanoseconds(c_local_time) };
+    // TODO(gitbuda): Check the i64 to u32 conversion.
+    let seconds = (c_nanoseconds / nsec_in_sec) as u32;
+    let nanoseconds = (c_nanoseconds % nsec_in_sec) as u32;
+    NaiveTime::from_num_seconds_from_midnight_opt(seconds, nanoseconds)
+}
+
+pub(crate) fn mg_value_naive_local_date_time(
+    mg_value: *const bindings::mg_value,
+) -> Option<NaiveDateTime> {
+    let c_local_date_time = unsafe { bindings::mg_value_local_date_time(mg_value) };
+    let c_seconds = unsafe { bindings::mg_local_date_time_seconds(c_local_date_time) };
+    let c_nanoseconds = unsafe { bindings::mg_local_date_time_nanoseconds(c_local_date_time) };
+    // TODO(gitbuda): Check the i64 to u32 conversion.
+    let nanoseconds_since_last_non_leap_sec = c_nanoseconds as u32;
```

Comment:
Instead of doing this conversion, why don't you construct two durations (one for c_seconds and one for c_nanoseconds)? Then construct midnight (all zeroed) NaiveLocalDateTime and add the second/nanosecond duration to it.

Reply:
I actually don't know how to do this elegantly because `NativeDate` / `NativeTime` don't expose a method to construct them from the number of seconds... Maybe I don't understand what the `mg_local_date_time_nanoseconds` is... Let's discuss that...